mvpa2/algorithms/searchlight_hyperalignment_surface.py

```diff
@@ -0,0 +1,564 @@
+                  "in the exclude list."% (params.ref_ds)            
+        if params.ref_ds != params.hyperalignment.params.ref_ds:
+            print('SHPAL: Supplied ref_ds & hyperalignment instance ref_ds:%d differ.'%(params.hyperalignment.params.ref_ds))
+            print('SHPAL: using default hyperalignment instance with ref_ds: %d'%(params.ref_ds))
+            params.hyperalignment = Hyperalignment(ref_ds=params.ref_ds)
+        if params.ref_ds < 0 and params.ref_ds >= ndatasets:
+            raise ValueError, "Requested reference dataset %i is out of " \
+                  "bounds. We have only %i datasets provided" \
+                  % (params.ref_ds, ndatasets)
+        if len(params.exclude_from_model) > 0:
+            print('SHPAL: These datasets will not participate in building common model:'),
+            print params.exclude_from_model
+        if params.voxsel_fnames is None or not np.alltrue([os.path.isfile(vxfn) for vxfn in params.voxsel_fnames]):
+            if params.pial_surf_fns is None or params.white_surf_fns is None or params.intermediate_surf_fns is None:
+                raise ValueError, "pial, white, & intermediate surface file names can't be None."\
+                    "They are required input arguments if pre-computed voxsel_fnames are given or present."
+            if len(params.pial_surf_fns) != ndatasets:
+                raise ValueError, "Length of pial_surf_fns %i doesn't match " \
+                    "the length of the datasets list %i."% (len(params.pial_surf_fns), ndatasets)
+            if len(params.white_surf_fns) != ndatasets:
+                raise ValueError, "Length of white_surf_fns %i doesn't match " \
+                    "the length of the datasets list %i."% (len(params.white_surf_fns), ndatasets)
+            if len(params.intermediate_surf_fns) != ndatasets:
+                raise ValueError, "Length of intermediate_surf_fns %i doesn't match " \
+                    "the length of the datasets list %i."% (len(params.intermediate_surf_fns), ndatasets)
+        
+        hmeasure = HyperalignmentMeasure(ndatasets=ndatasets, featsel=params.featsel, ref_ds=params.ref_ds,
+                      hyperalignment=params.hyperalignment, full_matrix=params.combine_neighbormappers,
+                      use_same_features=False, exclude_from_model=params.exclude_from_model, dtype=params.dtype)
+        # Zscoring time-series data if desired.
+        if params.zscore_all:
+            if __debug__:
+                debug('SHPAL', "Zscoring datasets")
+            for ds in datasets:
+                zscore(ds, chunks_attr=None)
+        
+        nfeatures = datasets[params.ref_ds].nfeatures
+        self.nfeatures_all = [sd.nfeatures for sd in datasets]
+        if params.voxsel_fnames is not None:
+            if len(params.voxsel_fnames) != ndatasets:
+                raise ValueError, "Length of voxsel_fnames %i doesn't match " \
```

Comment:
slowly moving from old style... use `raise ValueError(msg)` not `raise ValueError, msg`